registry/network/helpers_test.go

```diff
@@ -0,0 +1,49 @@
+package network
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestDetermineIPVersion tests the determineIPVersion function.
+func TestDetermineIPVersion(t *testing.T) {
+	nr := &NetworkRegistry{}
+
+	// Test case: IPv4 address
+	ipv4 := net.ParseIP("192.168.0.1")
+	assert.Equal(t, uint8(4), nr.determineIPVersion(ipv4))
```

Comment:
You could have used EqualValue, so without a need for conversion, but why not

Reply:
For me is more explicit than `EqualValue`